Octopus.Client/PowerShell/Tenants/UpdateTenantLogo.ps1

```diff
@@ -1,18 +1,124 @@
+        [string][parameter(Mandatory = $true)]$ApiKey
+    )
+    BEGIN
+    {
+        if (-not (Test-Path $InFile))
+        {
+            $errorMessage = ("File {0} missing or unable to read." -f $InFile)
+            $exception =  New-Object System.Exception $errorMessage
+			$errorRecord = New-Object System.Management.Automation.ErrorRecord $exception, 'MultipartFormDataUpload', ([System.Management.Automation.ErrorCategory]::InvalidArgument), $InFile
+			$PSCmdlet.ThrowTerminatingError($errorRecord)
+        }
+
+        if (-not $ContentType)
+        {
+            Add-Type -AssemblyName System.Web
+
+            $mimeType = [System.Web.MimeMapping]::GetMimeMapping($InFile)
+            
+            if ($mimeType)
+            {
+                $ContentType = $mimeType
+            }
+            else
+            {
+                $ContentType = "application/octet-stream"
+            }
+        }
+    }
+    PROCESS
+    {
+        Add-Type -AssemblyName System.Net.Http
+
+		$httpClientHandler = New-Object System.Net.Http.HttpClientHandler
+
+        $httpClient = New-Object System.Net.Http.HttpClient $httpClientHandler
+        $httpClient.DefaultRequestHeaders.Add("X-Octopus-ApiKey", $ApiKey)
+
+        $packageFileStream = New-Object System.IO.FileStream @($InFile, [System.IO.FileMode]::Open)
+        
+		$contentDispositionHeaderValue = New-Object System.Net.Http.Headers.ContentDispositionHeaderValue "form-data"
+	    $contentDispositionHeaderValue.Name = "fileData"
+		$contentDispositionHeaderValue.FileName = (Split-Path $InFile -leaf)
+
+        $streamContent = New-Object System.Net.Http.StreamContent $packageFileStream
+        $streamContent.Headers.ContentDisposition = $contentDispositionHeaderValue
+        $streamContent.Headers.ContentType = New-Object System.Net.Http.Headers.MediaTypeHeaderValue $ContentType
+        
+        $content = New-Object System.Net.Http.MultipartFormDataContent
+        $content.Add($streamContent)
+
+        try
+        {
+			$response = $httpClient.PostAsync($Uri, $content).Result
+
+			if (!$response.IsSuccessStatusCode)
+			{
+				$responseBody = $response.Content.ReadAsStringAsync().Result
+				$errorMessage = "Status code {0}. Reason {1}. Server reported the following message: {2}." -f $response.StatusCode, $response.ReasonPhrase, $responseBody
+
+				throw [System.Net.Http.HttpRequestException] $errorMessage
+			}
+
+			return $response.Content.ReadAsStringAsync().Result
+        }
+        catch [Exception]
+        {
+			$PSCmdlet.ThrowTerminatingError($_)
+        }
+        finally
+        {
+            if($null -ne $httpClient)
+            {
+                $httpClient.Dispose()
+            }
+
+            if($null -ne $response)
+            {
+                $response.Dispose()
+            }
+        }
+    }
+    END { }
+}
+
 # You can get this dll from your Octopus Server/Tentacle installation directory or from
```

Comment:
Hi @harrisonmeister
Thanks for this PR.
Could we please avoid recommending using the Octopus.Client.dll from the server installation folder? That would follow what is recommended on this [doc](https://octopus.com/docs/octopus-rest-api/octopus.client/getting-started#using-octopus.client-from-installation-folder).

We had [customers](https://octopus.zendesk.com/agent/tickets/87586) run into some trouble before.

Reply:
Added a PR [here](https://github.com/OctopusDeploy/OctopusDeploy-Api/pull/119) to fix. Would you mind reviewing @akirayamamoto ?